lib/internal/http2/util.js

```diff
@@ -156,24 +246,25 @@ function mapToHeaders(map) {
       continue;
     var isArray = Array.isArray(value);
     if (key[0] === ':') {
+      assertValidPseudoHeader(key);
       if (isArray) {
         if (value.length > 1)
-          throw new errors.Error('ERR_HTTP2_PSEUDO_HEADERS_SINGLE_VALUE');
+          throw new errors.Error('ERR_HTTP2_HEADER_SINGLE_VALUE', key);
         value = value[0];
       }
-      val = String(value);
-      assertIllegalConnectionSpecificHeader(key, val, mapToHeaders);
-      ret.unshift([key, val]);
+      ret.unshift([key, String(value)]);
     } else {
+      key = String(key).toLowerCase();
```

Comment:
Not quite. Header field names are case insensitive and must only be transmitted in hpack as lower case. Converting to lowercase would be perfectly acceptable.